### PR TITLE
blocks-gcs-proxy written in golang

### PR DESCRIPTION
**Pull request by akm**
This is the first version in Go Language.

## Difference from old version

- Source code
    - They're totally re-created.
- Configuration file
    - The file name is changed from `config.yml` to `config.json`
    - The structure is changed.
    - Environment variables are **NOT SUPPORTED YET** .
        - The next PR #20 includes environment variables support.
- Loggers are removed
- Use `gcloud deployment-manager` to build topics and subscriptions for debugging with pubsub.
- Version is bumped up from `0.3.2` to `0.4.0`

The `0.4.0` will be released after a few PRs are merged.

### CODE REVIEW

**Review by yodack — APPROVED** [review]

**Review by nagachika — APPROVED** [review]

File: .travis.yml [hunk 1]
```diff
@@ -1,5 +1,18 @@
...
+go:
+  - 1.6.3
+addons:
+  apt:
+    sources:
+      - sourceline: 'ppa:masterminds/glide'
+    packages:
+      - glide
+before_install:
+  ## Install glide
```
Comment (minimum2scp) [inline .travis.yml]: Unnecessary comment, glide is installed by addons section.

File: doc/how_it_works.md [hunk 2]
```diff
@@ -130,12 +138,15 @@ cmd1 path/to/workspace/uploads path/to/workspace/downloads/file1 path/to/workspa
...
-  b: gs://bucket1/file2
-  c: gs://bucket1/file3
+```json
+{
+  "foo": "key2",
+  "download_files": {
+    "a": "gs://bucket1/file1",
+    "b": "gs://bucket1/file2",
+    "c": "gs://bucket1/file3"
+  ]
```
Comment (minimum2scp) [inline doc/how_it_works.md]: `]` should be `}`.

File: .travis.yml (hunk 1 above)
Comment (akm) [reply .travis.yml]: @minimum2scp I removed it. Thank you.

File: doc/how_it_works.md (hunk 2 above)
Comment (akm) [reply doc/how_it_works.md]: @minimum2scp I fixed it. Thank you for always helping me.

**Review by minimum2scp — APPROVED** [review]

**Review by totty — APPROVED** [review]

**Review by taigou — APPROVED** [review]

**Review by hokuts — APPROVED** [review]

**Review by rumif — APPROVED** [review]

**Review by ko1nksm — APPROVED** [review]

**Review by guemon — APPROVED** [review]

**Pull request merged**